dataplane/handlers/interface.go

```diff
@@ -224,26 +224,40 @@ func (ni *Interface) handleLinkUpdate(lu *netlink.LinkUpdate) {
 	iface := ni.getOrCreateInterface(modelName)
 	iface.GetOrCreateEthernet().MacAddress = ygot.String(lu.Attrs().HardwareAddr.String())
 	iface.Ifindex = ygot.Uint32(uint32(lu.Attrs().Index))
-	iface.Enabled = ygot.Bool(lu.Attrs().Flags == net.FlagUp)
+	iface.Enabled = ygot.Bool(lu.Attrs().Flags&net.FlagUp != 0)
+	iface.AdminStatus = oc.Interface_AdminStatus_DOWN
+	if *iface.Enabled {
+		iface.AdminStatus = oc.Interface_AdminStatus_UP
+	}
+	// TODO: handle other states.
 	var operStatus oc.E_Interface_OperStatus
-	switch lu.Attrs().OperState { // TODO: handle other states.
+	switch lu.Attrs().OperState {
 	case netlink.OperDown:
 		operStatus = oc.Interface_OperStatus_DOWN
-	case netlink.OperUp:
+	case netlink.OperUp, netlink.OperUnknown: // TAP interface may be unknown state because the dataplane doesn't bind to its fd, so treat unknown as up.
```

Comment:
Interesting, so I guess the fact that the interface gets bound during port creation (via Build()) isn't at the right time, or isn't enough?

Reply:
i'm not sure what exactly cause the interface to be UP, but it works as long as its not down